nilearn/glm/tests/test_first_level.py

```diff
@@ -1831,6 +1832,16 @@ def test_check_trial_type_warning(tmp_path):
         _check_trial_type([event_file])
 
 
+def test_list_valid_subjects_with_toplevel_files(tmp_path):
+    """Test that only subject directories are returned, not file names."""
+    (tmp_path / "sub-01").mkdir()
+    (tmp_path / "sub-02").mkdir()
+    (tmp_path / "sub-01.html").touch()
+
+    valid_subjects = _list_valid_subjects(tmp_path, None)
+    assert valid_subjects == ["01", "02"]
+
+
```

Comment:
Adding this test because at one point the changes I had made caused the `visual_testing/reporter_visual_inspection_suite.py` script to fail during documentation building, and there wasn't a test to catch that error

Reply:
nice, thanks